### PR TITLE
Bump dask versions

### PR DESCRIPTION
**Pull request by galipremsagar**
This PR updates max version of dask & distributed.

Needed for : https://github.com/rapidsai/cudf/pull/8368

### CODE REVIEW

File: conda/recipes/versions.yaml [hunk 1]
```diff
@@ -47,11 +47,11 @@ cupy_version:
 cython_version:
   - '>=0.29.17,<0.30'
 dask_version:
-  - '=2021.4.0'
+  - '=2021.5.0'
```
Comment (kkraus14) [inline conda/recipes/versions.yaml:50]: Do we actually require this or can it be `>=2021.4.0,<=2021.5.0`?

Comment (galipremsagar) [reply conda/recipes/versions.yaml:50]: It can be `>=2021.4.0,<=2021.5.0`

**Review by kkraus14 — APPROVED** [review]

**Review by dillon-cullinan — APPROVED** [review]

**Comment by galipremsagar** [comment]
rerun tests

**Comment by jakirkham** [comment]
rerun tests

**Comment by dillon-cullinan** [comment]
Merged despite errors. Errors should be resolved by: https://github.com/rapidsai/cudf/pull/8368

**Comment by jakirkham** [comment]
Ah ok. Thanks Dillon! 😄

**Pull request merged**